packages/debug/src/browser/debug-session.tsx

```diff
@@ -298,10 +312,14 @@ export class DebugSession implements CompositeTreeElement {
             if (!await this.exited(1000)) {
                 await this.disconnect(restart);
             }
+        } else if (this.configuration.type === 'pwa-extensionHost') {
```

Comment:
Not sure what you mean. Could you elaborate?
What happens is that we enter twice this function when stopping hosted plugin - once for the "Launch Extension" thread and second time for the "Hosted Plugin" thread  - so first time it gets into 'pwa-extensionHost' part and second to last `else`.
What I couldn't find another solution to was the `this.fireExited()` line - without it it would not complete exiting the "Launch Extension" thread and not sure why (`'exited'` was not fired)

Reply:
why do you enter this function twice?